spec/services/exports/case_log_export_service_spec.rb

```diff
@@ -2,38 +2,77 @@
+  let(:expected_master_manifest_filename) { "Manifest_2022_02_08_0001.csv" }
+  let(:expected_master_manifest_filename2) { "Manifest_2022_02_08_0002.csv" }
+
+  let(:case_log) { FactoryBot.create(:case_log, :completed) }
 
   def replace_entity_ids(export_template)
     export_template.sub!(/\{id\}/, case_log["id"].to_s)
     export_template.sub!(/\{owning_org_id\}/, case_log["owning_organisation_id"].to_s)
     export_template.sub!(/\{managing_org_id\}/, case_log["managing_organisation_id"].to_s)
   end
 
-  context "when exporting case logs" do
+  context "when exporting daily case logs" do
     subject(:export_service) { described_class.new(storage_service) }
 
     let(:case_log) { FactoryBot.create(:case_log, :completed) }
 
     before do
       Timecop.freeze(case_log.updated_at)
+      allow(storage_service).to receive(:write_file)
     end
 
-    it "generate an XML export file with the expected filename" do
-      actual_filename = nil
-      allow(storage_service).to receive(:write_file) { |filename, _| actual_filename = filename }
-      export_service.export_case_logs
-      expect(actual_filename).to eq(expected_filename)
+    context "and no case logs is available for export" do
+      it "generates a master manifest with the correct name" do
+        expect(storage_service).to receive(:write_file).with(expected_master_manifest_filename, any_args)
+        export_service.export_case_logs
+      end
+
+      it "generates a master manifest with CSV headers but no data" do
+        actual_content = nil
+        expected_content = "zip-name,date-time zipped folder generated,zip-file-uri\n"
+        allow(storage_service).to receive(:write_file).with(expected_master_manifest_filename, any_args) { |_, arg2| actual_content = arg2&.string }
+
+        export_service.export_case_logs
+        expect(actual_content).to eq(expected_content)
+      end
     end
 
-    it "generate an XML export file with the expected content" do
-      actual_stringio = nil
-      allow(storage_service).to receive(:write_file) { |_, stringio| actual_stringio = stringio }
-      actual_content = replace_entity_ids(export_file.read)
-      export_service.export_case_logs
-      expect(actual_stringio&.string).to eq(actual_content)
+    context "and case logs are available for export" do
+      before do
+        case_log
+      end
+
+      it "generates an XML export file with the expected filename" do
+        expect(storage_service).to receive(:write_file).with(expected_data_filename, any_args)
+        export_service.export_case_logs
+      end
+
+      it "generates an XML export file with the expected content" do
+        actual_content = nil
+        expected_content = replace_entity_ids(export_file.read)
+        allow(storage_service).to receive(:write_file).with(expected_data_filename, any_args) { |_, arg2| actual_content = arg2&.string }
+
+        export_service.export_case_logs
+        expect(actual_content).to eq(expected_content)
```

Comment:
Wonder if this would be easier to test if we split the StringIO bit into a final step. So you can have a unit test just expect against the string value and then a second unit test check that string is passed into StringIO and written to StorageService correctly.

Not important for now though.

Reply:
I think that's what we are doing now, we have two tests, one for the filename and one for the stringio content.